File: Moxion.Presentation.Web/ClientApp/src/api/responses/KinematicSimulateResponse.ts
import ErrorCode from "../../models/enumerations/ErrorCode.ts";
import MotionProfile from "../../models/kinematics/MotionProfile.ts";
import MotionData from "../../models/kinematics/MotionData.ts";
import KinematicUnitInfo from "../../models/kinematics/units/KinematicUnitInfo.ts";

class KinematicSimulateResponse {
    errorCode: ErrorCode
    profile: MotionProfile | null
    data: MotionData[] | null
    unitInfo: KinematicUnitInfo | null

    constructor(
        errorCode?: ErrorCode,
        profile?: MotionProfile | null,
        data?: MotionData[] | null,
        unitInfo?: KinematicUnitInfo | null
    ) {
        this.errorCode = errorCode ?? 'NoError'
        this.profile = profile ?? null
        this.data = data ?? null
        this.unitInfo = unitInfo ?? null
    }
}

export default KinematicSimulateResponse

File: Moxion.Presentation/Features/Kinematic/Responses/KinematicSimulateResponse.cs
using Moxion.Common.Enumerations;
using Moxion.Domain.Units;
using Moxion.Presentation.Abstractions.Transport;
using Moxion.Presentation.Dto.Kinematic;

namespace Moxion.Presentation.Features.Kinematic.Responses;

public record KinematicSimulateResponse(
  ErrorCode ErrorCode,
  MotionProfileDto? Profile,
  MotionDataDto[]? Data,
  KinematicUnitInfo? UnitInfo
) : IResponse;